src/overlays/actors/ovl_En_Fish2/z_en_fish2.c

```diff
@@ -22,95 +65,1112 @@ const ActorInit En_Fish2_InitVars = {
+        Math_Vec3f_Copy(&sp2C, &this->unk_350->world.pos);
+        sp2C.x += Math_SinS(-0x3A98) * 10.0f;
+        sp2C.z += Math_CosS(-0x3A98) * 10.0f;
+        Math_Vec3f_Copy(&this->unk_2E8, &sp2C);
+    }
+
+    Play_CameraSetAtEye(globalCtx, this->unk_2CA, &this->unk_2E8, &this->unk_2DC);
+    if ((this->unk_350 == NULL) || (this->unk_350->update == NULL)) {
+        this->unk_350 = NULL;
+        this->unk_2B0++;
+        if (this->unk_2B0 > 10) {
+            this->unk_2B4 = 20;
+            this->actionFunc = func_80B2A23C;
+        }
+    }
+}
+
+void func_80B2A23C(EnFish2* this, GlobalContext* globalCtx) {
+    Vec3f sp2C;
+
+    Math_Vec3f_Copy(&sp2C, &this->actor.world.pos);
+    if (D_80B2B2E4 == 2) {
+        sp2C.x += (Math_SinS(-0x3A98) * 180.0f);
+        sp2C.y += 90.0f;
+        sp2C.z += Math_CosS(-0x3A98) * 180.0f;
+        Math_Vec3f_Copy(&this->unk_2DC, &sp2C);
+        Math_Vec3f_Copy(&sp2C, &this->actor.world.pos);
+        sp2C.y += 70.0f;
+        Math_Vec3f_Copy(&this->unk_2E8, &sp2C);
+    } else if (D_80B2B2F4 != NULL) {
+        Math_Vec3f_Copy(&sp2C, &D_80B2B2F4->world.pos);
+        sp2C.x += Math_SinS(-0x3A98) * 110.0f;
+        sp2C.z += Math_CosS(-0x3A98) * 110.0f;
+        Math_Vec3f_Copy(&this->unk_2DC, &sp2C);
+        Math_Vec3f_Copy(&sp2C, &D_80B2B2F4->world.pos);
+        sp2C.x += Math_SinS(-0x3A98) * 10.0f;
+        sp2C.z += Math_CosS(-0x3A98) * 10.0f;
+        Math_Vec3f_Copy(&this->unk_2E8, &sp2C);
+    }
+
+    Play_CameraSetAtEye(globalCtx, this->unk_2CA, &this->unk_2E8, &this->unk_2DC);
+
+    if ((this->unk_2B4 == 0) && (D_80B2B2E4 == 3)) {
+        D_80B2B2E0 = D_80B2B2EC = D_80B2B2E4 = 0;
+        D_80B2B2F4 = NULL;
+        ActorCutscene_Stop(this->unk_2BA[0]);
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80B2A448(EnFish2* this) {
+    func_80B28370(this, 4);
+    this->unk_2B4 = 0;
+    this->unk_2C4 = 0;
+    this->unk_2B6 = this->unk_2B4;
+    D_80B2B2E4 = 2;
+    this->actionFunc = func_80B2A498;
+}
+
+void func_80B2A498(EnFish2* this, GlobalContext* globalCtx) {
+    f32 currentFrame = this->skelAnime.animCurrentFrame;
+    Vec3f sp80;
+
+    if ((this->unk_2AC == 4) && func_801378B8(&this->skelAnime, 13.0f)) {
+        Actor* temp_v0;
+
+        Math_Vec3f_Copy(&sp80, &this->unk_318);
+        temp_v0 = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_COL_MAN, sp80.x, sp80.y, sp80.z, 0,
+                              this->actor.world.rot.y, 0, 0);
+        if (temp_v0 != NULL) {
+            temp_v0->speedXZ = 4.0f;
+            temp_v0->velocity.y = 15.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_SY_PIECE_OF_HEART);
+            gSaveContext.weekEventReg[81] &= 0xEF;
+            gSaveContext.weekEventReg[81] &= 0xDF;
+            gSaveContext.weekEventReg[81] &= 0xBF;
+            gSaveContext.weekEventReg[81] &= 0x7F;
+            gSaveContext.weekEventReg[82] &= 0xFE;
+            gSaveContext.weekEventReg[82] &= 0xFD;
```

Comment:
```suggestion
            gSaveContext.weekEventReg[81] &= ~0x10;
            gSaveContext.weekEventReg[81] &= ~0x20;
            gSaveContext.weekEventReg[81] &= ~0x40;
            gSaveContext.weekEventReg[81] &= ~0x80;
            gSaveContext.weekEventReg[82] &= ~0x1;
            gSaveContext.weekEventReg[82] &= ~0x2;
```
There's a chance you may need `(u8)` cast in front